src/renderer/components/+network-ingresses/ingress-class-menu.injectable.tsx

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import React from "react";
+import { getInjectable } from "@ogre-tools/injectable";
+import { withInjectables } from "@ogre-tools/injectable-react";
+import { computed } from "mobx";
+import type {
+  KubeObjectMenuItemComponent,
+  KubeObjectMenuItem,
+} from "../kube-object-menu/kube-object-menu-item-injection-token";
+import {
+  kubeObjectMenuItemInjectionToken,
+} from "../kube-object-menu/kube-object-menu-item-injection-token";
+import { ingressClassSetDefaultInjectable } from "./ingress-class-set-default.injectable";
+import { MenuItem } from "../menu";
+import type { IngressClass } from "../../../common/k8s-api/endpoints/ingress-class.api";
+import type { KubeObjectMenuProps } from "../kube-object-menu";
+import { Icon } from "../icon";
+
+export interface IngressClassMenuProps extends KubeObjectMenuProps<IngressClass> {
+  setDefault(item: IngressClass): void;
```

Comment:
nit: this should be a dependency and not a `Prop`. That way we shouldn't have to have the `as KubeObjectMenuItemComponent` below.

Reply:
Not sure if we can get rid of `as KubeObjectMenuItemComponent`.
I've extracted this prop to separated interface `Dependencies` and I still see the errors:

<img width="1101" alt="Screenshot 2023-01-05 at 18 00 49" src="https://user-images.githubusercontent.com/6377066/210825141-cce21b2b-8700-4e1b-8008-f73df273545d.png">

Other usages of `as KubeObjectMenuItemComponent`:

<img width="1008" alt="image" src="https://user-images.githubusercontent.com/6377066/210825369-8094d236-f097-4eb7-9a26-1939285824f4.png">

Reply:
Oh okay. I guess that makes sense from a typing perspective. Though it should be listed as a dep anyway.